hyperwallet/hyperwallet-ios-ui-sdk
# DTSERWONE-1091 Introduce Github Action

# Summary
PR to introduce GitHub Action to compare with TraviCI

## changes

- Introduce Git Hub Action to perform Unit and UI Tests
- Perform Strict Lint Check using `SwiftLint`
- Publish Test Coverage to Coveralls for all schemas
- Introduce fastlane task to perform Unit and UI Tests to get better logs
   - Use `fastlane ui_tests` to run UI tests
   - Use `fastlane unit_tests` to run Unit tests
- Disable animation during the UI Tests and Increase speed the tests navigation 
- Fix UI Tests by handling field navigation or enhancing how to find the find into the UI tests.
- Disable travis CI
- Add support to retry 2 attempts if UI Test fail for flaky
- Add GitHub Cache support to avoid to download and build Carthage Cache every commit. 
- Remove script to reset emulator before run UI Tests (No longer need it)

Note: On Github the UI and Unit test is taking ~55min, on TravisCI took ~1h40min

## Required
- [x] Perform Unit tests
- [x]  Lint check 
- [x] Report test coverage to Coveralls.io

## Result

### Last result from Travis CI 
![Screen Shot 2022-09-12 at 6 58 44 PM](https://user-images.githubusercontent.com/107439697/189790640-83a00fbb-f0fe-464c-8eca-faa9671241c4.png)

### Git Hub Action
[Git Hub Action result](https://github.com/grmeyer-hw-dev/hyperwallet-ios-ui-sdk/actions/runs/3041513449/usage)
![Screen Shot 2022-09-12 at 6 53 58 PM](https://user-images.githubusercontent.com/107439697/189790452-0c03543b-6dc1-4e21-88c1-9c144baf54fa.png)

### UI tests result
![Screen Shot 2022-09-12 at 6 53 00 PM](https://user-images.githubusercontent.com/107439697/189790955-7caec6fb-84f8-4fbb-b2ef-689e1005d41c.png)

### Unit tests result
![Screen Shot 2022-09-12 at 6 53 27 PM](https://user-images.githubusercontent.com/107439697/189790978-6a5373a1-2d85-48b9-829e-365155722d77.png)

### Code Coverage result
[Coverall result](https://coveralls.io/github/grmeyer-hw-dev/hyperwallet-ios-ui-sdk)
![Screen Shot 2022-09-12 at 7 08 20 PM](https://user-images.githubusercontent.com/107439697/189791797-8e430bc4-287d-4d57-9a8c-bc9ef6be6568.png)
